src/test/java/de/hpi/bpt/argos/eventProcessing/EventReceiverTest.java

```diff
@@ -147,44 +124,121 @@ public void testReceiveEvent_ChangeStatus_Success() throws Exception {
 		String newStatus = "NewStatus_" + ArgosTestUtil.getCurrentTimestamp();
 		testMapping.setTargetStatus(newStatus);
 		PersistenceAdapterImpl.getInstance().saveArtifacts(testMapping);
+		int eventsCount = getEventsCount();
 
 		WebSocket webSocket = WebSocket.buildWebSocket();
 
 		RestRequest request = RestRequestFactoryImpl.getInstance()
 				.createPostRequest(ARGOS_REST_HOST, getReceiveEventUri(testEventType.getId()));
 
-		JsonObject event = new JsonObject();
+		JsonObject event = getJsonEvent();
 
-		for (TypeAttribute eventTypeAttribute : testEventTypeAttributes) {
-			if (testEventType.getTimeStampAttributeId() == eventTypeAttribute.getId()) {
-				event.addProperty(eventTypeAttribute.getName(), ArgosTestUtil.getCurrentTimestamp());
-				continue;
+		setValidMapping(event);
+		request.setContent(serializer.toJson(event));
+		assertEquals(HttpStatusCodes.SUCCESS, request.getResponseCode());
+
+		assertEquals(eventsCount + 1, getEventsCount());
+		assertEquals(newStatus, getEntityStatus());
+
+		assertWebSocketMessages(webSocket);
+	}
+
+	@Test
+	public void testReceiveEvent_CustomMapping_Success() throws Exception {
+		Entity newEntity = ArgosTestUtil.createEntity(testEntityType, true);
+		String targetStatus = "TargetStatus_" + ArgosTestUtil.getCurrentTimestamp();
+		testMapping.setTargetStatus("UnreachableStatus_" + ArgosTestUtil.getCurrentTimestamp());
+		String testEntityStatus = getEntityStatus();
+		PersistenceAdapterImpl.getInstance().saveArtifacts(testMapping);
+		int eventsCount = getEventsCount();
+
+		EventCreationObserver customEventMapper = new EventCreationObserver() {
+			@Override
+			public void onEventCreated(EventEntityMappingStatus mappingStatus,
+									   EventType eventType,
+									   List<TypeAttribute> eventTypeAttributes,
+									   Event event,
+									   List<Attribute> eventAttributes) {
+
+				assertFalse(mappingStatus.isMapped());
+
+				mappingStatus.setEventOwner(newEntity);
```

Comment:
How do you know, the event isn't mapped to newEntity anyways, without this custom mapping?